azure-pipelines.yml

```diff
@@ -40,16 +40,36 @@ jobs:
 
   - script: |
       npm run test:ci || [ -f './test-results.xml' ]
-    displayName: 'Run tests (with JUnit output)'
+    displayName: 'Run CLI tests (with JUnit output)'
 
   - task: PublishTestResults@2
     inputs:
       testResultsFormat: 'JUnit'
       testResultsFiles: 'test-results.xml'
       searchFolder: '$(Build.SourcesDirectory)'
       failTaskOnFailedTests: true
-      testRunTitle: 'Unit tests'
-    displayName: 'Verify unit test result'
+      testRunTitle: 'CLI unit tests'
+    displayName: 'Verify CLI unit test result'
+
+  - script: |
+      npm ci
+      npm run build
+    displayName: 'Install and build appcenter-file-upload-client-node'
+    workingDirectory: appcenter-file-upload-client-node
+
+  - script: |
+      npm run test:ci || [ -f './test-results.xml' ]
+    displayName: 'Run appcenter-file-upload-client-node tests'
+    workingDirectory: appcenter-file-upload-client-node
+
+  - task: PublishTestResults@2
+    inputs:
+      testResultsFormat: 'JUnit'
+      testResultsFiles: 'test-results.xml'
+      searchFolder: '$(Build.SourcesDirectory)/appcenter-file-upload-client-node'
+      failTaskOnFailedTests: true
+      testRunTitle: 'appcenter-file-upload-client-node unit tests'
+    displayName: 'Verify appcenter-file-upload-client-node unit test result'
```

Comment:
Could we have an "install test" here? Where we validate that `npm install -g` works on the package? Ideally we should be verifying the scenario that didn't work (linux with sudo and no nvm), but I'm not sure this is easily feasible within this pipeline.

Reply:
We have npm install [here](https://github.com/microsoft/appcenter-cli/pull/1021/files#diff-fec826feae04e51c0d94076385408bdcR55), and if it doesn't work, then testing step will fail. I don't see similar verification steps for the `npm i` for CLI so not sure why we need it?

Reply:
Sorry, I meant installing the appcenter-cli package itself onto the system. i.e. `npm pack && npm install -g appcenter-cli*.tgz`.
What was failing in the previous attempt was installing this package on a user's computer.